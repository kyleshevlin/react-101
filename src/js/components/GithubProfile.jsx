import React from 'react'
import PropTypes from 'prop-types'
import GithubReposContainer from '../containers/GithubReposContainer'

const GithubProfile = ({ data }) =>
  data ? (
    <div className="github_profile">
      <div className="github_profile-name">
        <a href={data.url}>{data.name}</a>
      </div>
      <div className="github_profile-login">{data.login}</div>
      <div className="github_profile-bio">{data.bio}</div>
      <GithubReposContainer />
    </div>
  ) : null

GithubProfile.propTypes = {
  data: PropTypes.shape({
    login: PropTypes.string,
    name: PropTypes.string,
    url: PropTypes.string,
    bio: PropTypes.string
  })
}

GithubProfile.defaultProps = {
  data: null
}

export default GithubProfile
