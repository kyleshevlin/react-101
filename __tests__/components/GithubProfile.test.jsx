import React from 'react'
import GithubProfile from '../../src/js/components/GithubProfile'

describe('GithubProfile', () => {
  it('should be empty with no data', () => {
    const component = shallow(<GithubProfile />)

    expect(component).toMatchSnapshot()
  })

  it('should display UI with data', () => {
    const data = {
      url: 'foo.com',
      name: 'Kyle',
      login: 'kyle',
      bio: 'Cool bio!'
    }
    const component = shallow(<GithubProfile data={data} />)

    expect(component).toMatchSnapshot()
    expect(component.find('a').prop('href')).toEqual(data.url)
    expect(component.find('a').text()).toEqual(data.name)
    expect(component.find('.github_profile-login').text()).toEqual(data.login)
    expect(component.find('.github_profile-bio').text()).toEqual(data.bio)
  })
})
