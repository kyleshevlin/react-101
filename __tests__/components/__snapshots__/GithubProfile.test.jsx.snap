// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GithubProfile should be empty with no data 1`] = `""`;

exports[`GithubProfile should display UI with data 1`] = `
<div
  className="github_profile"
>
  <div
    className="github_profile-name"
  >
    <a
      href="foo.com"
    >
      Kyle
    </a>
  </div>
  <div
    className="github_profile-login"
  >
    kyle
  </div>
  <div
    className="github_profile-bio"
  >
    Cool bio!
  </div>
  <Connect(GithubReposContainer) />
</div>
`;
